packages/camera/camera/example/ios/RunnerTests/FLTSavePhotoDelegateTests.m

```diff
@@ -0,0 +1,122 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import AVFoundation;
+@import XCTest;
+#import <OCMock/OCMock.h>
+
+@interface FLTSavePhotoDelegateTests : XCTestCase
+
+@end
+
+@implementation FLTSavePhotoDelegateTests
+
+- (void)testHandlePhotoCaptureResult_mustSendErrorIfFailedToCapture {
+  NSError *error = [NSError errorWithDomain:@"test" code:0 userInfo:nil];
+  dispatch_queue_t ioQueue = dispatch_queue_create("test", NULL);
+  id mockResult = OCMClassMock([FLTThreadSafeFlutterResult class]);
+  FLTSavePhotoDelegate *delegate = [[FLTSavePhotoDelegate alloc] initWithPath:@"test"
+                                                                       result:mockResult
+                                                                      ioQueue:ioQueue];
+
+  [delegate handlePhotoCaptureResultWithError:error
+                            photoDataProvider:^NSData * {
+                              return nil;
+                            }];
+  OCMVerify(times(1), [mockResult sendError:error]);
+}
+
+- (void)testHandlePhotoCaptureResult_mustSendErrorIfFailedToWrite {
+  XCTestExpectation *resultExpectation =
+      [self expectationWithDescription:@"Must send IOError to the result if failed to write file."];
+  dispatch_queue_t ioQueue = dispatch_queue_create("test", NULL);
+  id mockResult = OCMClassMock([FLTThreadSafeFlutterResult class]);
+  OCMStub([mockResult sendErrorWithCode:@"IOError" message:@"Unable to write file" details:nil])
+      .andDo(^(NSInvocation *invocation) {
+        [resultExpectation fulfill];
+      });
+  FLTSavePhotoDelegate *delegate = [[FLTSavePhotoDelegate alloc] initWithPath:@"test"
+                                                                       result:mockResult
+                                                                      ioQueue:ioQueue];
+
+  // We can't use OCMClassMock for NSData because some XCTest APIs uses NSData (e.g.
+  // `XCTRunnerIDESession::logDebugMessage:`) on a private queue.
+  id mockData = OCMPartialMock([NSData data]);
+  OCMStub([mockData writeToFile:[OCMArg any] atomically:[OCMArg any]]).andReturn(NO);
```

Comment:
fyi you can use `OCMOCK_ANY` instead of `[OCMArg any]`, it's the same.  Up to you though.